test/val/val_id_test.cpp

```diff
@@ -3798,6 +3798,89 @@ OpFunctionEnd
           "ID 7 defined in block 6 does not dominate its use in block 9"));
 }
 
+TEST_F(ValidateIdWithMessage, SpecIdTargetNotSpecializationConstant) {
+  string spirv = kGLSL450MemoryModel + R"(
+OpDecorate %3 SpecId 200
+%1 = OpTypeVoid
+%2 = OpTypeFunction %1
+%int = OpTypeInt 32 0
+%3 = OpConstant %int 3
+%main = OpFunction %1 None %2
+%4 = OpLabel
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv.c_str());
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(
+      getDiagnosticString(),
+      HasSubstr("OpDecorate SpectId decoration target <id> '1' is not a "
```

Comment:
The reported ID looks funny.  It's 1 because the assembler assigns IDs sequentially from 1, based on which id name was mentioned first. Same thing happens in the later tests as well.

I'll patch this before merging in.